src/gdb_packet.c

```diff
@@ -138,35 +138,61 @@ int gdb_getpacket(char *packet, int size)
 	return i;
 }
 
+static void gdb_next_char(char c, unsigned char *csum)
```

Comment:
quite like this refactoring - it's a good call that might save some space if aggressively applied to the rest of gdb_packet.c